pylib/gyp/generator/compile_commands_json.py

```diff
@@ -108,10 +108,14 @@ def GenerateOutput(target_list, target_dicts, data, params):
         cwd = os.path.dirname(build_file)
         AddCommandsForTarget(cwd, target, params, per_config_commands)
 
+    output_dir = None
     try:
-        # generator_output can be `None` on Windows machines
-        output_dir = params["options"].generator_output or os.getcwd()
+        # generator_output can be `None` on Windows machines, or even not
+        # defined in other cases
+        output_dir = params["options"].generator_output
     except (AttributeError, KeyError):
+        pass
+    if output_dir is None:
         output_dir = params["generator_flags"].get("output_dir", "out")
```

Comment:
All other falsely values ("", (), {}, 0, 0.0, etc.) are equally dangerous as `None`
```suggestion
    output_dir = output_dir or params["generator_flags"].get("output_dir", "out")
```

Reply:
As you, though I wonder why would someone set it to be an empty dict or some other strange falsey value. 😛